### DISCUSSION TITLE
Using Apple App extensions

### DISCUSSION THREAD

**Discussion by danieleavitabile** [macvim-dev/macvim#1166] [Ideas]
I have seen that Apple now provides support for the so-called App Extensions https://developer.apple.com/app-extensions/ 
Do you think it may be possible, using these extensions, to invoke a macvim buffer when filling out, say, a form in Safari, or a bit of text in Keynote or Pages? This may be a way to go towards a much stronger Vim integration.

**Comment by ychin** [comment]
Hmm, so you would imagine a back-and-forth? Like using the extension to open MacVim, fill it in and save, then put the content back? I'm not 100% sure how feasible that is and would likely need to take a closer look.

**Reply by eirnym** [reply]
I see a few things that _might_ be interesting, but obviously it's a low priority. Potential candidates are: Quick Look Preview, Share, Xcode Source Editor (hehe, many people actually would like to use it) and Action (like "Edit this piece in MacVim")

**Comment by eirnym** [comment]
Honestly, Apple provides most of these things for a long time by now, and I'm happy to see that now all these APIs are public :)
